src/components/QueueBar/QueueBar.tsx

```diff
@@ -0,0 +1,134 @@
+import { ReactElement, useEffect, useMemo, useState } from 'react'
+import { Text } from '@gnosis.pm/safe-react-components'
+import Accordion from '@material-ui/core/Accordion'
+import AccordionSummary from '@material-ui/core/AccordionSummary'
+import AccordionDetails from '@material-ui/core/AccordionDetails'
+import IconButton from '@material-ui/core/IconButton'
+import ExpandMoreIcon from '@material-ui/icons/ExpandLessRounded'
+import CloseIcon from '@material-ui/icons/CloseRounded'
+import ClickAwayListener from '@material-ui/core/ClickAwayListener'
+import styled from 'styled-components'
+
+import { black300, screenSm } from 'src/theme/variables'
+import { TxsInfiniteScroll } from 'src/routes/safe/components/Transactions/TxList/TxsInfiniteScroll'
+import { TxLocationContext } from 'src/routes/safe/components/Transactions/TxList/TxLocationProvider'
+import { QueueTxList } from 'src/routes/safe/components/Transactions/TxList/QueueTxList'
+import { usePagedQueuedTransactions } from 'src/routes/safe/components/Transactions/TxList/hooks/usePagedQueuedTransactions'
+import { grey400, background } from 'src/theme/variables'
+
+type QueueBarProps = {
+  showQueueBar: boolean
+  setClosedBar: (close: boolean) => void
+}
+
+const QueueBar = ({ showQueueBar, setClosedBar }: QueueBarProps): ReactElement | null => {
+  const [expanded, setExpanded] = useState(false)
+
+  const collapseQueueBar = () => {
+    setExpanded((expanded) => !expanded)
+  }
+
+  const closeQueueBar = () => {
+    setClosedBar(true)
+    setExpanded(false)
+  }
+
+  const { isLoading, hasMore, next, transactions } = usePagedQueuedTransactions()
+
+  const queuedTxCount = useMemo(
+    () => (transactions ? transactions.next.count + transactions.queue.count : 0),
+    [transactions],
+  )
+
+  useEffect(() => {
+    if (queuedTxCount) {
+      setClosedBar(false)
+      setExpanded(false)
+    }
+  }, [queuedTxCount, setClosedBar])
+
+  if (!showQueueBar) {
+    return null
+  }
```

Comment:
I wouldn't pass this as a prop and instead conditionally render the QueueBar outside like so:
```
{showQueueBar & <QueueBar ... />}
```

Reply:
done!